com.palantir.typescript/src/com/palantir/typescript/preferences/ProjectPreferenceStore.java

```diff
@@ -202,6 +251,186 @@ private synchronized void insertValue(String name) {
+
+        IEclipsePreferences projectPreferences = this.getProjectPreferences();
+
+        for (Map.Entry<String, Object> tsConfigEntry : entries.entrySet()) {
+            if (tsConfigEntry.getValue() instanceof Map) {
+                decodeTsConfigEntries(jsonTreePath + tsConfigEntry.getKey() + ".", (Map) tsConfigEntry.getValue());
+            } else if (isSupportedTsConfigPath(jsonTreePath + tsConfigEntry.getKey())) {
+
+                String matchingPreference = TSCONFIG_PATH_TO_PREFERENCE.get(jsonTreePath + tsConfigEntry.getKey());
+
+                String value = tsConfigValueToPreferenceValue(tsConfigEntry.getValue(), matchingPreference);
+                TypeScriptPlugin.getDefault().getLog().log(new Status(IStatus.INFO, TypeScriptPlugin.ID,
+                    "setting preference " + matchingPreference + " to " + value));
+                setValue(matchingPreference, value);
+                projectPreferences.put(matchingPreference, value);
+            }
+        }
+    }
+
+    public String tsConfigValueToPreferenceValue(Object tsConfigValue, String matchingPreference) {
+        String value = null;
+        if (tsConfigValue != null) {
+            value = tsConfigValue.toString();
+
+            // TODO : awful mapping, we should rename our enums to match standard options / tsconfig values
+            if (matchingPreference.equals(IPreferenceConstants.COMPILER_TARGET)) {
+                value = value.toUpperCase();
+                if (TSCONFIG_TARGET_MAPPING.containsKey(value)) {
+                    value = TSCONFIG_TARGET_MAPPING.get(value);
+                }
+            } else if (matchingPreference.equals(IPreferenceConstants.COMPILER_MODULE)) {
+                value = value.toUpperCase();
+                if (TSCONFIG_MODULE_KIND_MAPPING.containsKey(value)) {
+                    value = TSCONFIG_MODULE_KIND_MAPPING.get(value);
+                }
+            } else if (matchingPreference.equals(IPreferenceConstants.COMPILER_MODULE_RESOLUTION)) {
+                value = value.toUpperCase();
+                if (TSCONFIG_MODULE_RESOLUTION_MAPPING.containsKey(value)) {
+                    value = TSCONFIG_MODULE_RESOLUTION_MAPPING.get(value);
+                }
+            } else if (matchingPreference.equals(IPreferenceConstants.COMPILER_JSX)) {
+                value = value.toUpperCase();
+            }
+        }
+        return value;
+    }
+
+    private boolean isTsConfigPreference(String name) {
+        return PREFERENCE_TO_TSCONFIG_PATH.containsKey(name);
+    }
+
+    private boolean isSupportedTsConfigPath(String tsConfigJsonPath) {
+        return TSCONFIG_PATH_TO_PREFERENCE.containsKey(tsConfigJsonPath);
+    }
+
+    private String getFileSHA1(IFile file) {
+
+        if (!file.exists()) {
+            return null;
+        }
+
+        InputStream fileInputStream = null;
+        try {
+            fileInputStream = file.getContents();
```

Comment:
I think you can try to use `Files.hash(file, Hashing.sha1())` from `com.google.common.hash`